### ISSUE TITLE
Accessible name missing for folder/file icons

### ISSUE THREAD

**Issue by eileencho** [jupyter/notebook#5029]
Description: The folder/file icons in the Notebook directory do not have accessible names. The icons contain important information that is not conveyed to screen readers. Screen reader users trying to navigate the directory have no information on whether the current item is a folder or a file.

Reproduce: 

1. Open Notebook dashboard to see the file directory
2. Download and start screen reader (such as [Voiceover](https://www.apple.com/voiceover/info/guide/_1124.html) which is already available on Macs, [NVDA](https://www.nvaccess.org/download/) on Windows, [Orca](https://wiki.gnome.org/Projects/Orca) for Linux)
3. Navigate with screen reader to the file directory
4. When selecting an item from file directory, note (by listening or reading the screen reader output) that the folder/file icon is skipped and each item declares its name (folder or file name) and role (link), but there is no information about whether the item is a folder or a file

Suggestions:
Make the folder/file icons visible to screen readers, and include alt text for the icons
Include information conveyed by the icon with the name of each item

[Example from NYU Accessibility Audit Supplement](https://docs.google.com/document/d/1q_7LiqLX2WGJZ0UHNWDFafw75McZgLp9v4Ifdzn2FYE/edit#heading=h.jromitgxrfr6)
[WCAG 2.0 Standard](https://www.w3.org/TR/UNDERSTANDING-WCAG20/ensure-compat-rsv.html)
[NYU Full Accessibility Audit](https://docs.google.com/spreadsheets/d/1iCiEExo4VDMOwAYtBTCdh2chyx394-1r6YwzmRILF9E/edit?usp=sharing) done with @VickySteeves

**Comment by balajialg** [comment]
@jtpio I tried testing the alpha version of Jupyter Notebook 7.0 using a [Screen reader tool](https://chrome.google.com/webstore/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn) available as a Chrome plugin. I tested this using a Chrome browser on a Windows machine. I found the above issue of screen readers not being able to convey the sub-menu information raised by @eileencho  still persists with the alpha version of Notebook 7.0. The screen reader shows blank responses when I hover over the icons present as a sub-menu. It reads the menu icon with text correctly  (like the interface button) but if there is a menu without text then the screen reader throws a blank response.

![image](https://user-images.githubusercontent.com/2306166/197597583-aad26948-c213-4bcb-a460-6bf6d0a4a180.png)

The menu icons appearing in the cells executing code also are not decipherable for the screen readers. 
![image](https://user-images.githubusercontent.com/2306166/197597717-c432a957-1114-4ea9-90d8-abe9eaf5cfe5.png)

I guess this should get addressed and prioritized for the upcoming Notebook 7.0 release.  Let me know if you need more information. Thanks!
